src/ui/DrawButton.js

```diff
@@ -0,0 +1,14 @@
+import { useState, useCallback } from "react";
+
+export default function DrawButton({ onClick }) {
+  const [HasDrawn, setHasDrawn] = useState(false);
+
+  const checkClick = async () => {
+    setHasDrawn(!HasDrawn);
+    await onClick();
+  };
+
+  return (
+    <button onClick={checkClick}>{HasDrawn ? "New Game?" : "Draw!"}</button>
```

Comment:
This is good. Do you know Nathaniel? He was asking how to do if expressions in React yesterday. Would be cool if you would share this example with him.